masonry/src/doc/04_testing_widget.md

```diff
@@ -12,4 +12,197 @@
 
 </div>
 
-TODO
\ No newline at end of file
+<!-- Not compleletey confident about this joke. Mighty want to put an expiration date on it. -->
+<!-- On the other hand, I'm curious how long it stays relevant. -->
+The test harness in Masonry is designed to provide a robust and flexible environment for writing and executing tests for your Rust GUI applications and-
+
+Okay, you know what, ChatGPT isn't great at this doc thing.
+Let's start over.
+
+A week before writing this document's first draft, this author had a chat with a friend, who complained about the difficulties of writing tests for native GUI apps.
+
+His reasoning went that you could always encapsulate your business logic into pure functions and test their inputs and outputs, but you had no way to write "When I click on this button this widget should appear" into your codebase in a way CI could enforce mechanically.
+
+This one anecdote aside, wanting to test your UI in a headless setting is something many developers want, hence the proliferations of tools testing tools around the WebDriver standard for web dev.
+Giving similar affordances was a core design goal of Masonry.
```

Comment:
I'd consider a much more to-the-point introduction. Something like:
```suggestion
Masonry supports testing your UI in a headless setting, with similar capabilities to testing tools which use the WebDriver standard for web dev.
```
The provided anecdote is equivalent to saying "no other cross-platform GUI library supports testing", which I know to be false. We should be happy for other tools to also allow tests to be written.

Reply:
Yeah. There's a deeper truth in there, something like "Even when GUI frameworks support testing it's often under-advertised and people don't know about it", but I guess documentation isn't the right place to soapbox about that.

Reply:
I do like your version. It's concise and to the point.